frontend/dockerfile/dockerfile_test.go

```diff
@@ -1546,18 +1609,27 @@ ARG BAR=${FOO:?"foo missing"}
 }
 
 func testMultiArgs(t *testing.T, sb integration.Sandbox) {
-	integration.SkipOnPlatform(t, "windows")
 	f := getFrontend(t, sb)
 
-	dockerfile := []byte(`
+	dockerfile := []byte(integration.UnixOrWindows(
+		`
 ARG a1="foo bar" a2=box
 ARG a3="$a2-foo"
 FROM busy$a2 AS build
 ARG a3 a4="123 456" a1
 RUN echo -n "$a1:$a3:$a4" > /out
 FROM scratch
 COPY --from=build /out .
-`)
+`,
+		`
+ARG a1="foo bar" a2=server
+ARG a3="$a2-foo"
+FROM nano$a2 AS build
+USER ContainerAdministrator
+ARG a3 a4="123 456" a1
+RUN echo %a1%:%a3%:%a4%> /out
```

Comment:
We should keep the "scratch" stage imo:

```suggestion
RUN echo %a1%:%a3%:%a4%> /out
FROM nanoserver
COPY --from=build /out .
```

Reply:
We currently have an outstanding issue with multi-stage builds for WCOW, see https://github.com/moby/buildkit/issues/5193
So, I was suggesting just simplifying the test as long as it captures the intent?